### PR TITLE
DSC19-78: Case Outcome

### PR DESCRIPTION
**Pull request by de-laz**

### CODE REVIEW

File: input/fsh/who-co-Observations.fsh [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+Instance: WhoCoEncounterExample
+InstanceOf: WhoCoEncounter
+Usage: #example
+* location.location = Reference(WhoCrLocationExample)
+* status = #finished
+* class = http://terminology.hl7.org/CodeSystem/v3-ActCode#ACUTE
+* reasonCode = WhoCrCodeSystemReasonForTesting#CASE_CONTACT
+* period.start = "2021-05-12"
+* period.end = "2021-05-13"
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: Please move these Encounter resources into their own file.
Having each resource in its own file makes finding and maintaining the profiles a bit easier.

File: input/fsh/who-co-QuestionnaireResponse.fsh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+* item[=].text = "Results of last test"
+* item[=].answer[+].valueCoding = WhoCoPositiveNegativeUnknown#positive
+
+* item[+].linkId = "outcome_contacts_followed"
+* item[=].text = "Total number of contacts followed for this case"
+* item[=].answer[+].valueInteger = 5
+
+* item[+].linkId = "outcome_contacts_followed_unknown"
+* item[=].text = "Number of contacts followed unknown"
+* item[=].answer[+].valueCoding = $cs-v2-0136#UNK
```
Comment (MattyJ007) [inline input/fsh/who-co-QuestionnaireResponse.fsh]: Missing newline at end of file

File: input/fsh/who-co-composition.fsh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* section[=].entry[+] = Reference(WhoCoObservationOutcomePatcourseEcmoExample)
+* section[=].entry[+] = Reference(WhoCoObservationOutcomePatcourseStatusExample)
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeLabDateExample)
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeLabResultExample)
+* section[=].entry[+] = Reference(WhoCoHospitalizationEncounterExample)
+
+* section[+].title = "exposure risk"
+* section[=].code = http://test.org/sectionCode#exposureRisk
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeContactsFollowedUnknownExample)
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeContactsFollowedExample)
```
Comment (MattyJ007) [inline input/fsh/who-co-composition.fsh]: Missing newline at end of file

File: input/fsh/who-co-Observations.fsh [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+* status = #final
+* valueCodeableConcept = WhoCoPositiveNegativeUnknown#positive
+
+Instance: WhoCoObservationOutcomeContactsFollowedExample
+InstanceOf: WhoCoObservationOutcomeContactsFollowed
+Usage: #example
+Title: "WHO CO Outcome Contacts Followed Example"
+Description: "Total number of contacts followed for this case"
+* status = #final
+* valueInteger = 5
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh:171]: @de-laz Could you please add an example where the integer value is -1 and include an `Observation.note: "unknown number of contacts follwed"`
This is just to show how we want `unknown` to be shown for this observation.

File: input/fsh/who-co-Observations.fsh [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+Instance: WhoCoObservationOutcomePatcourseStatusExample
+InstanceOf: WhoCoObservationOutcomePatcourseStatus
+Usage: #example
+Title: "WHO CO Outcome Patcourse Status Example"
+Description: "Health Outcome"
+* status = #final
+* valueCodeableConcept = WhoCoHealthOutcome#death
+
+Instance: WhoCoObservationOutcomeLabDateExample
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: Please add this date value to the observation below as an `effectiveDateTime`

File: input/fsh/who-co-Observations.fsh [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+
+Instance: WhoCoObservationPatcourseAdmitExample
+InstanceOf: WhoCoObservationPatcourseAdmit
+Usage: #example
+Title: "WHO Case Outcome Patcourse Admit Example"
+Description: "Admission to hospital (may have been previously reported)"
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
+
+Instance: WhoCoObservationPatcourseAdmitDateExample
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: Please add this date value to the Observation above as an `effectiveDateTIme`.
Maybe you could also include an Observation in the negative (which would then exclude the date).
We won't necessarily use it in the `Bundle` but it'll be good to show the alternative form of the Observation.

File: input/fsh/who-co-Observations.fsh [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+
+Instance: WhoCoObservationOutcomeAsympExample
+InstanceOf: WhoCoObservationOutcomeAsymp
+Usage: #example
+Title: "WHO CO Outcome Asymp Example"
+Description: "Developed symptoms after time of specimen collection"
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
+
+Instance: WhoCoObservationOutcomeAsympDateExample
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: Please add this date value to the Observation above as an `effectiveDateTIme`.
Maybe you could also include an Observation in the negative (which would then exclude the date).
We won't necessarily use it in the Bundle but it'll be good to show the alternative form of the Observation.

File: input/fsh/who-co-Observations.fsh [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+
+Profile: WhoCoObservationOutcomeContactsFollowedUnknown
+Parent: Observation
+Id: who-co-observation-outcome-contacts-followed-unknown
+Title: "WHO CO Outcome Contacts Followed Unknown"
+Description: "WHO CO Outcome Contacts Followed Unknown Observation"
+* code = $LNC#96552-5
+* value[x] only CodeableConcept
+* valueCodeableConcept 0..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: This observation can be deleted

File: input/fsh/who-co-Observations.fsh [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+* value[x] only dateTime
+* valueDateTime 0..1
+
+Profile: WhoCoObservationOutcomeLabResult
+Parent: Observation
+Id: who-co-observation-outcome-lab-result
+Title: "WHO CO Outcome Lab Result"
+Description: "WHO CO Outcome Lab Result Observation"
+* code = $LNC#96552-5
+* value[x] only CodeableConcept
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh:79]: This Observation gets `effectiveDateTIme`

File: input/fsh/who-co-Observations.fsh [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+Profile: WhoCoObservationOutcomePatcourseStatusOther
+Parent: Observation
+Id: who-co-observation-outcome-patcourse-status-other
+Title: "WHO CO Outcome Patcourse Status Other"
+Description: "WHO CO Outcome Patcourse Status Other Observation"
+* value[x] only string
+* valueString 0..1
+
+Profile: WhoCoObservationOutcomeLabDate
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: This observation can be removed as its data will be included below

File: input/fsh/who-co-Observations.fsh [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+Parent: Observation
+Id: who-co-observation-patcourse-admit
+Title: "WHO Case Outcome Patcourse Admit"
+Description: "WHO Case Outcome Patcourse Admit Observation"
+* code = $LNC#77974-4
+* value[x] only CodeableConcept
+* valueCodeableConcept 1..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
+
+Profile: WhoCoObservationPatcourseAdmitDate
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: This Observation can be removed as the data will be recorded above

File: input/fsh/who-co-Observations.fsh [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+* value[x] only dateTime
+* valueDateTime 0..1
+
+Profile: WhoCoObservationPatcourseAdmit
+Parent: Observation
+Id: who-co-observation-patcourse-admit
+Title: "WHO Case Outcome Patcourse Admit"
+Description: "WHO Case Outcome Patcourse Admit Observation"
+* code = $LNC#77974-4
+* value[x] only CodeableConcept
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh:19]: Add effectiveDateTIme

```suggestion
* value[x] only CodeableConcept
* effective[x] only DateTime
```

Not sure if my syntax is correct

File: input/fsh/who-co-Observations.fsh [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+Parent: Observation
+Id: who-co-observation-outcome-asymp
+Title: "WHO Case Outcome Outcome Asymp"
+Description: "WHO Case Outcome Outcome Asymp Observation"
+* code = $LNC#66421-9
+* value[x] only CodeableConcept
+* valueCodeableConcept 1..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
+
+Profile: WhoCoObservationOutcomeAsympDate
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: This Observation can be removed - data is included above

File: input/fsh/who-co-Observations.fsh [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+* period 1..1
+* period.end 1..1
+
+Profile: WhoCoObservationOutcomeAsymp
+Parent: Observation
+Id: who-co-observation-outcome-asymp
+Title: "WHO Case Outcome Outcome Asymp"
+Description: "WHO Case Outcome Outcome Asymp Observation"
+* code = $LNC#66421-9
+* value[x] only CodeableConcept
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh:7]: Add effectiveDateTime

File: input/fsh/who-co-Observations.fsh [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+* period 1..1
+* period.end 1..1
+
+Profile: WhoCoEncounter
+Parent: WhoCrEncounter
+Id: who-co-encounter
+Title: "WHO CO Encounter"
+Description: "WHO CO Encounter"
+* period 1..1
+* period.end 1..1
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: These should be in an Encounter file to make it easier to maintain

File: input/fsh/who-co-Observations.fsh [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+* #unknown
+  "Unknown"
+* #other
+  "Other"
+
+ValueSet: WhoCoValueSetHealthOutcome
+Id: who-co-health-outcome
+Title: "Health Outcome"
+Description: "WH Co Valueset Health Outcome"
+* codes from system WhoCoHealthOutcome
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: Should these maybe go in the existing terminology file? Just to keep all related structure definitions together and to allow for reuse.
I'm not sure...
@rcrichton

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
@de-laz Looking good - just a few updates to the spec to cater to

File: input/fsh/who-co-Observations.fsh (hunk 16 above)
Comment (de-laz) [reply input/fsh/who-co-Observations.fsh]: @MattyJ007 do you think they should be in the existing file or a new file specific to `Case Outcome`?

Comment (MattyJ007) [reply input/fsh/who-co-Observations.fsh]: In the existing file should be fine - these options seem generic

File: input/fsh/who-co-Observations.fsh [hunk 17]
```diff
@@ -111,6 +111,8 @@ Description: "WHO CO Outcome Contacts Followed Observation"
 * code = $LNC#96551-7
 * value[x] only integer
 * valueInteger 1..1
+* note 0..*
+* note.text 1..1
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: These definitions seem unnecessary as they are the same as the original spec
https://www.hl7.org/fhir/observation.html#resource

File: input/fsh/who-co-Observations.fsh [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+Parent: Observation
+Id: who-co-observation-outcome-lab-result
+Title: "WHO CO Outcome Lab Result"
+Description: "WHO CO Outcome Lab Result Observation"
+* code = $LNC#96552-5
+* value[x] only CodeableConcept
+* valueCodeableConcept 0..1
+* valueCodeableConcept from WhoCoValueSetPositiveNegativeUnknown
+* effective[x] only dateTime
+* effectiveDateTime 1..1
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]:
```suggestion
* effectiveDateTime 0..1
```
In the spec this value won't always be supplied

File: input/fsh/who-co-Observations.fsh [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+* valueCodeableConcept = $cs-v2-0136#Y
+* effectiveDateTime = "2021-05-13"
+
+Instance: WhoCoObservationOutcomeAsympNoEffectiveDateExample
+InstanceOf: WhoCoObservationOutcomeAsymp
+Usage: #example
+Title: "WHO CO Outcome Asymp Example"
+Description: "Developed symptoms after time of specimen collection example without effective date"
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
```
Comment (MattyJ007) [inline input/fsh/who-co-Observations.fsh]: In a more realistic case, I think the asymp date is only left out when the patient says "no".
But this is pedantic

```suggestion
Instance: WhoCoObservationOutcomeAsympNoEffectiveDateExample
InstanceOf: WhoCoObservationOutcomeAsymp
Usage: #example
Title: "WHO CO Outcome Asymp Example"
Description: "Did not develop symptoms after time of specimen collection example (no effective date)"
* status = #final
* valueCodeableConcept = $cs-v2-0136#N
```

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
Looking great 👍🏼 
Just some very minor things

**Review by MattyJ007 — APPROVED** [review]
👍🏼

**Pull request merged**